overlord/snapstate/autorefresh.go

```diff
@@ -873,11 +887,90 @@ func maybeAddRefreshInhibitNotice(st *state.State) error {
 		}
 	}
 
-	if changed {
-		if _, err := st.AddNotice(nil, state.RefreshInhibitNotice, "-", nil); err != nil {
+	if err := maybeAddRefreshInhibitWarningFallback(st, curInhibitedSnaps, changed); err != nil {
+		return err
+	}
+
+	if !changed {
+		// the set of inhibited snaps was not changed, nothing to do.
+		return nil
+	}
+
+	if _, err := st.AddNotice(nil, state.RefreshInhibitNotice, "-", nil); err != nil {
+		return err
+	}
+	st.Set("last-recorded-inhibited-snaps", curInhibitedSnaps)
+
+	return nil
+}
+
+// maybeAddRefreshInhibitWarningFallback records a warning if the set of
+// inhibited snaps was changed since the last notice.
+//
+// The warning is recorded only if:
+// 	1. There is at least 1 inhibited snap.
+// 	2. The "refresh-app-awareness-ux" experimental flag is enabled.
+// 	3. No snap exists with the marker "snap-refresh-observe" interface connected.
+func maybeAddRefreshInhibitWarningFallback(st *state.State, inhibitedSnaps map[string]bool, changed bool) error {
+	if len(inhibitedSnaps) == 0 {
```

Comment:
shouldn't we potentially remove the warning in that case?

Reply:
good point, I introduced `RemoveWarning`